### PR TITLE
Task 13. Friends App

### PR DESCRIPTION
**Pull request by babayK0**
### Task 13. Friends App

[Demo](https://babayk0.github.io/Friends-app/) |
[Code base](https://github.com/babayK0/Friends-app)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/babayK0/Friends-app/scripts.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
+const FRIENDS_API = "https://randomuser.me/api/?results=30";
+const GET_FRIENDS = fetch(FRIENDS_API);
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: Redundant variable.

File: submissions/babayK0/Friends-app/scripts.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+const RESET = document.querySelector(".reset");
+
+let users = [];
+
+GET_FRIENDS.then(response => response.json()).then(json => {
+    users = json.results;
+    makeFriend(users);
+});
+
+function makeFriend(users) {
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: This is a very large function. Pls, split it.

File: submissions/babayK0/Friends-app/scripts.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		INFO_CONTAINER.appendChild(MAIL);
+
+		const PHONE = document.createElement('div');
+		PHONE.innerHTML=`Call me ! `
+		+`<a href="tel:${element.phone}" title="${element.phone}">${ICONS.phone}</a>`;
+		PHONE.classList.add('phone');
+		INFO_CONTAINER.appendChild(PHONE);
+
+		const HOME = document.createElement('p');
+		HOME.innerHTML=(`${element.location.city}`).toUpperCase()
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: `text-transform: uppercase`

File: submissions/babayK0/Friends-app/scripts.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+	CONTENT.appendChild(CONTENT_FRAGMENT);
+}
+
+SORT_AGE.addEventListener("click", function ({target}){
+	switch(target.value){
+		case "age-asc":
+			users.sort(function(a,b){
+				return a.dob.age - b.dob.age;
+			})
+			showSorted(users);
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: `showSorted` can be moved outside `switch` statement.

Also, this only two conditions here. You can use a simple `if` statement.

File: submissions/babayK0/Friends-app/scripts.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+}
+SORT_NAME.addEventListener("click", function ({target}){
+	switch(target.value){
+		case "name-az":
+			users = sortName(users);
+			showSorted(users);
+			break;
+		case "name-za":
+			users = sortName(users);
+			showSorted(users.reverse());
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: Good idea with `reverse`. But keep in mind that this is an additional operation.

File: submissions/babayK0/Friends-app/scripts.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+		case "man":
+			showSorted(users);
+			break;
+		case "woman":
+			showSorted(users);
+			break;
+		case "all":
+			showSorted(users);
+			break;
+	}
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: It is 3 checks with the same result. Why you need them?

File: submissions/babayK0/Friends-app/scripts.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+});
+
+SEARCH.addEventListener("input",function({target}){
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: And you need `target` here for what?

File: submissions/babayK0/Friends-app/scripts.js [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+	showSorted(users);
+});
+
+function showSorted(users){
+	let sortedUsers = [];
+	users.forEach(function(element){
+		if(!`${element.name.first}${element.name.last}`
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: Can you move this check into a separate variable with an understandable name?

File: submissions/babayK0/Friends-app/scripts.js [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+			}
+		}
+		if(ALL.checked){
+			if(element.gender==="female"||element.gender==="male"){
+				sortedUsers.push(element);
+				return sortedUsers;
+			}
+		}		
+	});
+	console.log(sortedUsers);
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: You should remove `console.log` prior PR.

File: submissions/babayK0/Friends-app/scripts.js [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+	CONTENT.innerHTML='';
+	makeFriend(sortedUsers);
+}
+
+RESET.addEventListener("click",function({target}){
+	console.log(target);
+	let radio = Array.prototype.slice.call(RADIO);
```
Comment (zonzujiro) [inline submissions/babayK0/Friends-app/scripts.js]: `Array.prototype.slice.call` - why?

File: submissions/babayK0/Friends-app/scripts.js (hunk 5 above)
Comment (babayK0) [reply submissions/babayK0/Friends-app/scripts.js]: @zonzujiro okay

**Comment by babayK0** [comment]
@zonzujiro done :smile:

**Comment by babayK0** [comment]
priziwaju @zonzujiro

**Pull request merged**